### PR TITLE
Refactor SimpleObjectUnionScalarEvaluator

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/jsonitem/SimpleObjectUnionScalarEvaluator.java [hunk 1]
```diff
@@ -24,79 +24,88 @@
...
+        TaggedValuePointable tempTvp = ppool.takeOne(TaggedValuePointable.class);
+        TaggedValuePointable tempValue = ppool.takeOne(TaggedValuePointable.class);
+        try {
+            abvs.reset();
+            ob.reset(abvs);
+            tvps.clear();
+            if (arg.getTag() == ValueTag.SEQUENCE_TAG) {
+                arg.getValue(sp);
                 for (int i = 0; i < sp.getEntryCount(); ++i) {
                     op = (ObjectPointable) ObjectPointable.FACTORY.createPointable();
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/jsonitem/SimpleObjectUnionScalarEvaluator.java:61]: Add a check to make sure the item is an object.

**Comment by prestoncarman** [comment]
Please add one test case: when two objects have the same key. Then I think this change is ready to go.

**Comment by riyafa** [comment]
These changes seem to be already in master. Shall I create a new PR with the tests?

**Comment by prestoncarman** [comment]
Yes, Please.

**Pull request closed without merging**